crates/cxx-qt-lib-extras/src/core/qcommandlineoption.rs

```diff
@@ -0,0 +1,112 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Laurent Montel <laurent.montel@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use cxx::{type_id, ExternType};
+use std::mem::MaybeUninit;
+
+#[cxx::bridge]
+mod ffi {
+    unsafe extern "C++" {
+        include!("cxx-qt-lib-extras/qcommandlineoption.h");
+        type QCommandLineOption = super::QCommandLineOption;
+        include!("qstring.h");
```

Comment:
i would keep this `cxx-qt-lib/qstring.h`.

Then instead in the build.rs add `cxx-qt-lib`, eg make -extras depend on -lib (so you'll need the build and dependency in the cargo.toml too)

```rust
    builder
        .with_opts(cxx_qt_lib_headers::build_opts())
        .with_opts(cxx_qt_lib_extras_headers::build_opts())
        .build();
```

Reply:
ah ok cool. I thought that some code was missing :) thanks